src/App.tsx

```diff
@@ -1,11 +1,97 @@
-import { useState } from 'react';
+import {
+  ChangeEvent,
+  FormEvent,
+  MouseEvent,
+  useState,
+} from 'react';
+
 import './App.scss';
+
+import {
+  BASE_IMDB_URL,
+  getMovie,
+  emptyImg,
+} from './api';
+
+import { Movie } from './types/Movie';
+import { MovieData } from './types/MovieData';
+
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
-import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState(false);
+
+  const onChangeQuery = (e: ChangeEvent<HTMLInputElement>) => {
+    if (isError) {
+      setError(false);
+    }
+
+    setQuery(e.target.value);
+  };
+
+  const onFindMovie = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (movie) {
+      setMovie(null);
+    }
+
+    setLoading(true);
+
+    try {
+      const request = await getMovie(query);
+
+      if ('Error' in request) {
+        setError(true);
```

Comment:
maybe we can check it with `!request.ok`?

Reply:
![image](https://user-images.githubusercontent.com/118932043/223750261-de66530c-7338-4b75-afe4-ad05b96175dd.png)